### PR TITLE
CRUD methods for bookController and BookNotFoundException

### PR DESCRIPTION
**Pull request by alanlamaswolox**
## Summary
Created CRUD methods and BookNotFoundException

## Trello card
[link](https://trello.com/c/tlxJ7lhM/18-crear-las-operaciones-crud-de-book)

### CODE REVIEW

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/BookController.java:16]: A good practice mentioned at this point in the training is to add java docs in each of the controllers. They are also used to describe and document methods that may be considered complex.

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) throws BookNotFoundException {
+        bookRepository.findById(id)
+            .orElseThrow(() -> new BookNotFoundException(id.toString(), "id"));
+        bookRepository.deleteById(id);
+    }
+
+    @PostMapping
+    public void create(@RequestBody Book book){
+        bookRepository.save(book);
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/BookController.java]: Add a book return to show the book you added at the end of the transaction

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Good job, leave you two little comments 😄

**Review by AleFalcon — DISMISSED** [review]
It's okay with comments by Nico

File: src/main/java/wolox/training/controllers/BookController.java [hunk 3]
```diff
@@ -1,13 +1,67 @@
...
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    /**
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/BookController.java:17]: One of the most important things about javadocs is the explanation of the method being documented.
That description would be missing, indicating the purpose of the method in each of the documented methods.

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by AleFalcon — APPROVED** [review]

**Pull request merged**